### PR TITLE
add confusable non-ascii variables to weird-exprs.rs

### PR DESCRIPTION
**Pull request by betseg**

### CODE REVIEW

File: src/test/ui/weird-exprs.rs [hunk 1]
```diff
@@ -7,6 +7,8 @@
 #![allow(dead_code)]
 #![allow(unreachable_code)]
 #![allow(unused_braces, unused_must_use, unused_parens)]
```
Comment (lukaslueg) [inline src/test/ui/weird-exprs.rs:9]: The `allow` could be local to `confusable_non_ascii`, not global to the module, so it doesn't inadvertently affect other code and the point of `confusable_non_ascii` is more clear.

Comment (betseg) [reply src/test/ui/weird-exprs.rs:9]: > `warning: allow(confusable_idents) is ignored unless specified at crate level`

Comment (lukaslueg) [reply src/test/ui/weird-exprs.rs:9]: 😒

**Comment by Mark-Simulacrum** [comment]
What is the intent behind adding this? It seems like if anything it should go into https://github.com/rust-lang/rust/blob/master/src/test/ui/lint/rfc-2457-non-ascii-idents/lint-confusable-idents.rs, but I'm not sure this test is really adding value to what is already there. Perhaps you can elaborate on the purpose here?

**Comment by crlf0710** [comment]
@betseg Ping from triage, could you address the comments above?

**Pull request closed without merging**